Lib/test/test_argparse.py

```diff
@@ -3603,9 +3603,10 @@ class TestHelpUsage(HelpTestCase):
           -h, --help            show this help message and exit
           -w W [W ...]          w
           -x [X ...]            x
-          --foo, --no-foo       Whether to foo
+          --foo, --no-foo       Whether to foo (default: None)
```

Comment:
IMO showing `None` here is just as confusing as `SUPRESS`. Because, I mean, from the tool user's perspective, what does it mean?

Reply:
Just like any other argument, I believe conveying the meaning to the user is the responsibility of the description:
```
--uid                     The user id to assign to the user.  If None, then let the
                          system assign the next available value. (default: None)
--hide, --no-hide-user    Whether to omit the user from the graphical system login.
                          If None, then use the system default (default: None)
```

Reply:
However, I think this is a good example of why I mentioned that BooleanOptionalArgument adding defaults should be revisited.  The argparse architecture gives the programmer the ability to use a formatter_class.  If you want to display default values for your arguments but not if they were `None`, then you can write your own HelpFormatter to do that.  Since BooleanOptionalArguments modifies the help string on its own, not using a formatter class, the programmer cannot use argparse's standard features to choose different cases where the default should not be emitted.

Removing this formatting feature from BooleanOptionalArguments gives the programmer the flexibility to choose a formatter that does what they decide is appropriate.  Leaving it in means that you'll always be stuck with someone having to work around the choices that BooleanOptionalArguments makes.

Reply:
abadger: it may be better then to avoid displaying default value because the author of the help string can integrate it into the help, if he feels it's needed. In other words, when help message is not provided, it will be less confusing; if the author feels it's important enough to explain the effect of special value, they will test the output and see that default value needs to be integrated into the help. In either case, confusion will be avoided.